plugin/pkg/scheduler/factory/factory.go

```diff
@@ -307,6 +510,63 @@ func (c *ConfigFactory) updateNodeInCache(oldObj, newObj interface{}) {
 	if err := c.schedulerCache.UpdateNode(oldNode, newNode); err != nil {
 		glog.Errorf("scheduler cache UpdateNode failed: %v", err)
 	}
+
+	c.invalidateCachedPredicatesOnNodeUpdate(newNode, oldNode)
+}
+
+func (c *ConfigFactory) invalidateCachedPredicatesOnNodeUpdate(newNode *v1.Node, oldNode *v1.Node) {
+	if c.enableEquivalenceClassCache {
+		// Begin to update equivalence cache based on node update
+		invalidPredicates := sets.NewString()
```

Comment:
This seems to be the only place where you need it. But please add a TODO to lazily initialize it.